analysis_driver/quality_control/calculate_relatedness.py

```diff
@@ -46,4 +55,126 @@ def run_vcftools(self):
+                openfile.write('\t'.join(line))
+        return ped_file
+
+    def family_id(self, sample_id):
+        family_id = clarity.get_sample(sample_id).udf.get('Family ID')
+        if not family_id:
+            return 'No_ID'
+        return family_id
+
+    def relationship(self, member):
+        relationship = clarity.get_sample(member).udf.get('Relationship')
+        if not relationship:
+            return 'Other'
+        return relationship
+
+    def sex(self, member):
+        sex = clarity.get_sample(member).udf.get('Sex')
+        if not sex:
+            return 'No_Sex'
+        return sex
+
+    def relationships(self, family):
+        relationship_codes = {'Proband': {'Mother':'0', 'Father':'0'},
+                              'Mother':{'Mother':'0', 'Father':'0'},
+                              'Father':{'Mother':'0', 'Father':'0'},
+                              'Sister':{'Mother':'0', 'Father':'0'},
+                              'Brother': {'Mother':'0', 'Father':'0'},
+                              'Other':{'Mother':'0', 'Father':'0'}}
+
+        for member in family:
+            relationship = self.relationship(member)
+            if relationship in ['Father', 'Mother']:
+                    for i in ['Proband', 'Sister', 'Brother']:
+                        relationship_codes[i][relationship] = member
+        return relationship_codes
+
+    def get_member_details(self, family, all_families):
+        family_lines = []
+        family_info = self.relationships(all_families[family])
+        for member in all_families[family]:
+            sex_codes = {'Male': '1', 'Female': '2', 'No_Sex': '0'}
+            relationship = self.relationship(member)
+            line = [family,
+                    member,
+                   family_info[relationship]['Father'],
+                   family_info[relationship]['Mother'],
+                   sex_codes[self.sex(member)],
+                   '0']
+            family_lines.append(line)
+        return family_lines
+
+    @property
+    def ped_file_content(self):
+        all_families = {}
+        for i in self.ids:
+            family_id = self.family_id(i)
+            if not all_families.get(family_id):
+                all_families[family_id] = []
+            all_families[family_id].append(i)
+
+        ped_file_content = []
+        for family in all_families:
+            family_lines = self.get_member_details(family, all_families)
+            for line in family_lines:
+                ped_file_content.append(line)
+        return ped_file_content
+
+    @property
+    def peddy_command(self):
+        ped_file = self.write_ped_file()
+        peddy_cmd = 'peddy --plot --prefix %s %s %s' % (self.dataset.name, self.gatk_outfile + '.gz', ped_file)
+        return peddy_cmd
+
+    def run_peddy(self):
+        return executor.execute(
+            self.peddy_command,
+            job_name='peddy',
+            working_dir=self.job_dir,
+            cpus=10,
```

Comment:
Also tabix and bgzip above are being run in a single thread and use very little memory, so don't need 12 cpus and 30GB.